### PR TITLE
Added digamma function for TensorFlow frontend

### LINKED ISSUE #23001: digamma

### PR DESCRIPTION
**Pull request by Thathsarani-Sandarekha**
# Add Digamma Function and Tests to Ivy Tensorflow Math Library

## Summary:
This pull request introduces the Digamma function to the Ivy math library and adds corresponding unit tests. The Digamma function calculates the digamma (Ψ) function, which is a commonly used special function in mathematical and statistical computations.

Close #23001

## Changes Made

- [ ] Did you add a function?

Added the Digamma function to math.py with the following details:

* Function name: digamma(x, name=None)
* Description: Computes the digamma function.
* Arguments:
x (array): Input array.
name (str): Name for the operation (optional).
* Returns: An array containing the result of the digamma function applied to the input.
* Raises: NotImplementedError if the backend is not TensorFlow.
* Implementation: The function checks the backend, and if it's TensorFlow, it computes the digamma function using Ivy and returns the result.
- [ ] Did you add the tests?

Introduced unit tests for the Digamma function in test_tensorflow_math.py with the following details:

* Test function name: test_tensorflow_digamma
* Test decorator: @handle_frontend_test
* Tested operation: tensorflow.math.digamma

- [ ] Did you follow the steps we provided?

Yes

### Reason for Changes:

The addition of the Digamma function is essential for various mathematical and statistical computations. By including this function in Ivy's math library and providing comprehensive unit tests, we ensure its correctness and usability, enhancing the overall functionality of the library.

These changes contribute to Ivy's capabilities and support for mathematical operations, making it a more versatile and reliable library for scientific computing.

Please review and consider merging these changes to enhance Ivy's functionality and maintain its compatibility with TensorFlow and other backends.

### CODE REVIEW

**Comment by Thathsarani-Sandarekha** [comment]
Hi, @jshepherd01! Could you please review my PR? Thank you!

File: .idea/runConfigurations/_template__of_py_test.xml [hunk 1]
```diff
@@ -8,7 +8,6 @@
       <env name="PYTHONPATH" value="/opt/fw/numpy:/opt/fw/jax:/opt/fw/tensorflow:/opt/fw/torch:/opt/fw/paddle:/opt/fw/mxnet" />
     </envs>
     <option name="SDK_HOME" value="python" />
-    <option name="SDK_NAME" value="Remote Python 3.10.0 Docker (unifyai/ivy:latest)" />
```
Comment (jshepherd01) [inline .idea/runConfigurations/_template__of_py_test.xml:11]: Could you explain why you made this change?

File: .idea/runConfigurations/_template__of_py_test.xml [hunk 2]
```diff
@@ -50,4 +49,4 @@
     <option name="_new_targetType" value="&quot;PYTHON&quot;" />
     <method v="2" />
   </configuration>
-</component>
+</component>
```
Comment (jshepherd01) [inline .idea/runConfigurations/_template__of_py_test.xml:52]: It looks like you've introduced a formatting issue here (all files should end with newline characters). Please revert this change

File: ivy/data_classes/array/layers.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # global
 import abc
-from typing import Optional, Tuple, Union, List, Sequence, Dict
+from typing import Optional, Tuple, Union, List, Sequence
```
Comment (jshepherd01) [inline ivy/data_classes/array/layers.py:3]: I don't see what this has to do with your changes, was this from the pre-commit?

File: ivy/functional/frontends/tensorflow/math.py [hunk 4]
```diff
@@ -226,6 +227,15 @@ def cumsum(x, axis, exclusive=False, reverse=False, name=None):
     )
 
 
+@to_ivy_arrays_and_back
+def digamma(x, name=None):
+    backend = _get_backend(x)
+    if backend == "tensorflow":
+        return ivy.array(ivy.digamma(x), ivy.array([0], ivy.int32))
+    else:
+        raise NotImplementedError(f"digamma is not implemented for backend: {backend}")
```
Comment (jshepherd01) [inline ivy/functional/frontends/tensorflow/math.py:236]: Please don't do this. If the `ivy.digamma` function only does the right calculation with the tensorflow backend, please raise an issue to let us know - it should be the same regardless of backend.

Also, could you explain the composition you have here? Why not simply `return ivy.digamma(x)`?

File: ivy/functional/frontends/torch/comparison_ops.py [hunk 5]
```diff
@@ -290,7 +290,7 @@ def topk(input, k, dim=None, largest=True, sorted=True, *, out=None):
 
 
 gt = greater
+ne = not_equal
 ge = greater_equal
 le = less_equal
 lt = less
-ne = not_equal
```
Comment (jshepherd01) [inline ivy/functional/frontends/torch/comparison_ops.py:296]: I'm not sure why you made this change but it's causing lint errors, please revert it.

File: ivy/functional/frontends/tensorflow/math.py [hunk 6]
```diff
@@ -11,6 +11,7 @@
     handle_tf_dtype,
     to_ivy_dtype,
 )
+from ivy.framework_handler import get_backend as _get_backend
```
Comment (jshepherd01) [inline ivy/functional/frontends/tensorflow/math.py:14]: This import is causing the following error in testing:
```
E   ModuleNotFoundError: No module named 'ivy.framework_handler'
```

**Review by jshepherd01 — CHANGES_REQUESTED** [review]
As well as these comments, you also seem to have moved a few functions in various files. Was that from running the pre-commit hook?

File: ivy/functional/frontends/tensorflow/math.py (hunk 6 above)
Comment (Thathsarani-Sandarekha) [reply ivy/functional/frontends/tensorflow/math.py:14]: @jshepherd01 Really sorry for my mistakes. I have made a new PR resolving all these issues. I did not change anything in the other files it happened because I ran pre-commit in my local pc. Is that okay to delete this branch?

**Comment by Thathsarani-Sandarekha** [comment]
Added updated digamma function for TensorFlow frontend #23114 
Hence I am closing this PR.
